pkg/resources/smtp.go

```diff
@@ -19,17 +19,16 @@ type alertManagerConfig struct {
 	Global map[string]string `yaml:"global"`
 }
 
-func GetExistingSMTPFromAddress(ctx context.Context, client k8sclient.Client, operatorNs string) (string, error) {
+func GetExistingSMTPFromAddress(ctx context.Context, client k8sclient.Client, ns string) (string, error) {
 	amSecret := &corev1.Secret{}
-	err := client.Get(ctx, types.NamespacedName{Name: alertManagerConfigSecretName,
-		Namespace: operatorNs}, amSecret)
-
+	err := client.Get(ctx, types.NamespacedName{
+		Name:      alertManagerConfigSecretName,
+		Namespace: ns,
+	}, amSecret)
 	if err != nil {
 		return "", err
 	}
-
 	monitoring := amSecret.Data[alertManagerConfigSecretFileName]
```

Comment:
There could possible be a blank value here. Can error checking be added to on the existence of the `alertManagerConfigSecretFileName`. A test case should be added to for the missing data key. 

As github wont allow me to add the comment I will put it here. The final return statement could also have an error if the `stmp_from` key is missing. We should be testing for this also.